modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/resources/manifest/DefaultClassPackageFinder.java

```diff
@@ -32,14 +23,11 @@
   private List<ClassPackageFinder> classFinders = new ArrayList<>();
 
   public DefaultClassPackageFinder() {
-    classFinders.add(this::usingClassloader);
+    this.classFinders.add(new ClassloaderClassPackageFinder());
   }
 
-  public DefaultClassPackageFinder(ProcessingEnvironment processingEnvironment) {
-    this();
-    if (processingEnvironment != null) {
-      classFinders.add(className -> usingProcessingEnvironment(className, processingEnvironment));
-    }
+  public void add(ClassPackageFinder classPackageFinder) {
```

Comment:
I would rename the method to something more meaningful, "additionalPackageFinder()` or some other name.
Also, add some JDOC